tests/test_helper.py

```diff
@@ -280,3 +337,151 @@ def test_get_usd_cent_equivalent(mocker):
+
+    hlp.write_csv_output([
+        ['bitcoin', '2010-01-01', False, False, 'absolute', 0, False, False, 100],
+        ['ethereum', '2010-01-01', False, False, 'absolute', 0, False, False, 200],
+    ])
+    with open(pathlib.Path(__file__).resolve().parent / 'output-no_clustering-exclude_contract_addresses-absolute_10-exclude_below_fees-exclude_below_usd_cent.csv') as f:
+        lines = f.readlines()
+        assert lines[0] == ','.join(['ledger', 'snapshot_date', 'clustering', 'exclude_contract_addresses',
+                                     'top_limit_type', 'top_limit_value', 'exclude_below_fees',
+                                     'exclude_below_usd_cent', 'hhi']) + '\n'
+        assert lines[1] == ','.join(['bitcoin', '2010-01-01', 'False', 'False', 'absolute', '0', 'False', 'False',
+                                     '100']) + '\n'
+        assert lines[2] == ','.join(['ethereum', '2010-01-01', 'False', 'False', 'absolute', '0', 'False', 'False',
+                                     '200']) + '\n'
+    os.remove(pathlib.Path(__file__).resolve().parent / 'output-no_clustering-exclude_contract_addresses-absolute_10-exclude_below_fees-exclude_below_usd_cent.csv')
+
+
+def test_get_active_source_keywords(mocker):
+    get_config_mock = mocker.patch("tokenomics_decentralization.helper.get_config_data")
+    get_config_mock.return_value = {'analyze_flags': {'clustering_sources': ['test']}}
+
+    active_sources = hlp.get_active_source_keywords()
+    assert active_sources == ['test']
+
+
+def test_get_active_sources(mocker):
+    json_load_mock = mocker.patch('json.load')
+    json_load_mock.return_value = {'Test 1': ['test1', 'test11'], 'Test 2': ['test2']}
+
+    get_active_source_keywords_mock = mocker.patch("tokenomics_decentralization.helper.get_active_source_keywords")
+    get_active_source_keywords_mock.return_value = ['Test 1']
+
+    active_sources = hlp.get_active_sources()
+    assert active_sources == set(['test1', 'test11'])
+
+
+def test_get_clusters(mocker):
+    mocker.patch('builtins.open', mocker.mock_open(
+        read_data='{"name": "entity1", "address": "addr1", "source": "test"}\n{"name": "entity2", "address": "addr2", "source": "test"}\n{"name": "entity3", "address": "addr2", "source": "test2"}\n{"name": "entity3", "address": "addr1", "source": "test2"}\n{"name": "entity4", "address": "addr4", "source": "test"}\n{"name": "entity5", "address": "addr4", "source": "test2"}\n{"name": "entity6", "address": "addr6", "source": "test"}\n{"name": "entity7", "address": "addr1", "source": "test3"}'
+    ))
+
+    get_active_sources_mock = mocker.patch("tokenomics_decentralization.helper.get_active_sources")
+    get_active_sources_mock.return_value = ['test', 'test2']
+
+    clusters = hlp.get_clusters('bitcoin')
+    assert clusters['entity1'] == clusters['entity2']
+    assert clusters['entity1'] == clusters['entity3']
+    assert clusters['entity4'] == clusters['entity5']
```

Comment:
Also worth testing that the inactive sources (like test3) were not used, eg by asserting that entity7  is not in the clusters